hyperspy/io_plugins/fei.py

```diff
@@ -580,13 +619,53 @@ def ser_reader(filename, objects=None, verbose=False, *args, **kwds):
         'mapping': mapping}
     return dictionary
 
-
-def get_mode(mode):
+def guess_units_from_mode(objects_dict, header, verbose=False):
+    # assuming that for an image stack, the UnitsLength of the "3rd" dimension
+    # is 0
+    isImageStack = (header['Dim-1_UnitsLength'][0] == 0)
+    # in case the xml file doesn't contains the "Mode" return "Unknow" unit
+    try:
+        mode = objects_dict.ObjectInfo.ExperimentalDescription.Mode
+    except AttributeError: # in case the xml chunk doesn't contain the Mode
+        return 'Unknown'
+    isCamera = ("CameraNamePath" in objects_dict.ObjectInfo.AcquireInfo.keys())
+    # Workaround: if this is not an image stack and not a STEM image, then we 
+    # assume that it should be a diffraction
+    isDiffractionScan = (header['Dim-1_DimensionSize'][0] > 1 and not isImageStack)
+    if verbose:
+        print "------------"
+        print objects_dict.ObjectInfo.AcquireInfo
+        print "mode", mode
+        print "isCamera:", isCamera
+        print "isImageStack:", isImageStack
+        print "isImageStack:", isDiffractionScan
+        print "------------"
+    if 'STEM' in mode:
+        # data recorded in STEM with a camera, so we assume, it's a diffraction
+        if isCamera:
+            return "1/meters"
+        # in case we can't make use the detector is a camera, use a workaround
+        elif isDiffractionScan:
+            return "1/meters"
+        else:
+            return "meters"
+    elif 'Diffraction' in mode:
+        return "1/meters"
+    elif 'Image' in mode:
+        return "meters"
+    else:
+        return 'Unknown'
```

Comment:
Same as above: currently it would set the units to "meters" which is correct in most cases, and when it is not (diffraction) it is obvious that it is incorrect, so it is not a big deal. Return meters then?